lib/SampleService/core/config.py

```diff
@@ -198,24 +239,89 @@ def _check_string_req(s: Optional[str], name: str) -> str:
 }
 
 
-def get_validators(url: str) -> MetadataValidatorSet:
+def get_validators(
+    repo_path: Optional[str] = None, 
+    repo_asset: Optional[str] = None, 
+    tag: Optional[str] = None,
+    prerelease_ok: bool = False, 
+    url: Optional[str] = None, 
+    token: Optional[str] = None
+    ) -> MetadataValidatorSet:
     '''
-    Given a url pointing to a config file, initialize any metadata validators present
-    in the configuration.
+    Given a github repo or url pointing to a config file, initialize any metadata 
+    validators present in the configuration. repo_path or url must be specified.
+    If a github repo is specified, pull config from the release asset repo_asset.
 
+    :param repo_path: github repo path (i.e. kbase/sample_service_validator_config)
+    :param repo_asset: release asset filename containing validator config
+    :param tag: Tag specifying a specific release to load the config from (deafult is latest)
+    :param prerelease_ok: If false, ignores pre-releases when pulling validators from github
     :param url: The URL for a config file for the metadata validators.
+    :param token: The token to be used when querying github
     :returns: A set of metadata validators.
     '''
     # TODO VALIDATOR make validator CLI
+
     try:
-        with _urllib.request.urlopen(url) as res:
-            cfg = _yaml.safe_load(res)
+        config_asset = None
+        if url:
+            config_url = url
+        elif not repo_path:
+            raise ValueError(f'No metadata validator config URL or repo path.')
+        else:
+            if not repo_asset:
+                raise ValueError(f'No repo_asset name provided for repo "{repo_path}"')
+            try: 
+                repo = _Github(login_or_token=token).get_repo(repo_path)
+                releases = [rel for rel in repo.get_releases() if prerelease_ok or not rel.prerelease]
+            except Exception as e:
+                raise RuntimeError(f'Fetching releases from repo "{repo_path}" failed.') from e
+
+            if not releases:
+                raise ValueError(f'No releases found in validator config repo "{repo_path}"')
+
+            if tag:
+                target_release = next((r for r in releases if r.tag_name==tag), None)
+                if not target_release:
+                    raise ValueError(f'No release with tag "{tag}" found in validator config repo "{repo_path}"')
+            else:
+                # Use the latest release
+                target_release = releases[0]
+
+            assets = target_release.get_assets()
+            if not assets:
```

Comment:
what is "this"?

Reply:
If you mean the no-assets case. It's not currently replicable with GH, but the API spec doesnt specifically disallow an empty array here so, this is a "just in case" error that would require mocking GH reponses to test, which i think would be wild overkill

Reply:
If it's practically impossible for this case to occur, I would just take code path out, add a comment that currently google will always have > 0 assets. If things change it'll throw a `TypeError` which I think is fine